rust/src/nfs/nfs.rs

```diff
@@ -653,15 +653,21 @@ impl NFSState {
     }
 
     pub fn process_write_record<'b>(&mut self, r: &RpcPacket<'b>, w: &Nfs3RequestWrite<'b>) -> u32 {
-        // for now assume that stable FILE_SYNC flags means a single chunk
-        let is_last = if w.stable == 2 { true } else { false };
-
         let mut fill_bytes = 0;
-        let pad = w.file_len % 4;
+        let pad = w.count % 4;
         if pad != 0 {
             fill_bytes = 4 - pad;
         }
 
+        if w.count == 0 || w.count < w.file_len ||
+            ((fill_bytes + w.count) as usize) < w.file_data.len() ||
```

Comment:
Since at this stage we have already established that `count >= file_len`, and in the next check, we have `(fill_bytes + w.file_len) >= w.file_data.len()`, would this be an extra check?